newdle/schemas.py

```diff
@@ -44,29 +53,47 @@ def sort_users(self, data, many, **kwargs):
 
 
 class NewUnknownParticipantSchema(mm.Schema):
+    id = fields.Integer()
     name = fields.String(required=True)
 
 
+def validate_signature(data, **kwargs):
+    data = dict(data)
+    signature = data.pop('signature', None)
+    if not signature or not check_user_signature(
+        data, signature, fields=('name', 'email', 'auth_uid')
+    ):
+        raise ValidationError("Participant's user signature is invalid!")
+
+
 class NewKnownParticipantSchema(NewUnknownParticipantSchema):
     email = fields.String(required=True)
     auth_uid = fields.String(required=True)
     signature = fields.String(required=True)
 
     @validates_schema
-    def validate_signature(self, data, **kwargs):
-        data = dict(data)
-        signature = data.pop('signature')
-        auth_uid = data.pop('auth_uid')
-        if not check_user_signature(dict(data, uid=auth_uid), signature):
-            raise ValidationError("Participant's user signature is invalid!")
+    def validate_participant(self, data, **kwargs):
+        validate_signature(data, **kwargs)
```

Comment:
why the `**kwargs`?

Reply:
It was the 'not sure if we might need it'. But there's absolutely no point now.